app/models/energy/ScheduleBase.py

```diff
@@ -0,0 +1,100 @@
+"""Schedule Base Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class ScheduleContinuous(BaseModel):
+    """This Numeric Type allows all numbers, including fractional amounts, within the range to
+  be valid."""
+
+    type: Enum('ScheduleContinuous', {'type': 'ScheduleContinuous'})
+
+    schedule_continuous: float = None
+
+
+class ScheduleDiscrete(BaseModel):
+    """This Numeric Type allows all only integers within the range to be valid."""
+
+    type: Enum('ScheduleDiscrete', {'type': 'ScheduleDiscrete'})
+
+    schedule_discrete: int = None
+
+class ScheduleNumericType (BaseModel):
+    """Designates how the range values are validated."""
+
+    numeric_type:  Union[ScheduleContinuous, ScheduleDiscrete]
+
+
+class ScheduleUnitType (str, Enum):
+    dimensionless = 'Dimensionless'
+    temperature = 'Temperature'
+    delta_temperature = 'DeltaTemperature'
+    precipitation_rate = 'PrecipitationRate'
+    angle = 'Angle'
+    convection_coefficient = 'ConvectionCoefficient'
+    activity_level = 'ActivityLevel'
+    velocity = 'Velocity'
+    capacity = 'Capacity'
+    power = 'Power'
+    availability = 'Availability'
+    percent = 'Percent'
+    control = 'Control'
+    mode = 'Mode'
+
+
+class Date(BaseModel):
+    """Date."""
+
+    month: int = Schema(
+        1,
+        description='A value for month between `1`-`12`. Default is `1`.',
+        ge=1,
+        le=12
+    )
+
+    day: int = Schema(
+        1,
+        description='A value for day between `1`-`31`. Default is `1`.',
+        ge=1,
+        le=31
+    )
+
+
+class Time(BaseModel):
```

Comment:
I suggest moving this definition to the `app/models/common/datetime.py` file

Reply:
See my comment above.  It was aspirational to have a common datetime object but, after thinking about it more, we are really going to need separate for daylight and energy.

Reply:
I agree with @macumber, datetime object should be in a separate file and not in the Schedule module. This is different from the discussion that we had about using them between daylight and energy model schedules.